docs/introduction/tasks.md

```diff
@@ -1,148 +1,238 @@
+# input arguments would create a new and distinct task
+# task2 = splishsplash_simulator.run(input_dir="splishsplash-example",
+#                                  sim_config_filename="config.json")
+# print (task2.id)  # outputs k9muu1vq1fc6m2oyxm0n3n8y0 (example id)
 
-# Kill a task that hasn't completed yet.
-if status == "submitted" or status == "started":
-    task.kill()
 ```
 
+A task is uniquely identified by an alphanumeric identifier which is unique across
+all tasks submitted to the API by the user. However, a user can create multiple
+`Task` objects that point to the same task by instantiating the `Task` class with
+the same identifier without this resulting in the creation of new tasks on the
+API; they will be different Python objects, but they will refer to the same task
+on the API. This mechanism is useful when the user wants to recreate a `Task`
+object to retrieve information about tasks created in previous sessions.
+
 ```python
+>>> import inductiva
+>>>
+>>> task1 = inductiva.tasks.Task("i4ir3kvv62odsfrhko4y8w2an")
+>>> task2 = inductiva.tasks.Task("i4ir3kvv62odsfrhko4y8w2an")
+>>> print(id(task1))
+4410160112 
+>>> print(id(task2))
+4389863104
+>>> print(task1.id)
+i4ir3kvv62odsfrhko4y8w2an
+>>> print(task2.id)
+i4ir3kvv62odsfrhko4y8w2an  # the same as task1.id
+```
 
-# Block the call until the simulation completes, but
-# without interrupting the remote task if the local process
-# is interrupted. 
-task.wait()
+### Managing a task
 
-# Or turn the simulation into a blocking call. Users kill
-# the remote process if the local process is killed.
-with task.sync_context():
-    task.wait()
-```
+The `Task` class provides mechanisms to track and manage the status of a task.
+With a `Task` object at hand, the user can:
+ * Get the status, i.e., whether the task is running, has finished, has failed, etc;
+ (see section about [task lifecycle](#task-lifecycle) for more details);
+ * Get the machine type where it ran/is running;
+ * Get the execution time.
+ * Download either all or just a subset of the output files produced by the task;
+ * Turning it into a blocking call;
+ * Kill it;
 
+The following snippets show how to use the `Task` class to manage a task.
 
 ```python
+# create a new task or retrieve a task from a previous session
+# task = simulator.run(...)
+#   or 
+# task = Task(i4ir3kvv62odsfrhko4y8w2an)
 
-# If you don't know the files generated by your task, you can get an object
-# representing information on the output files.
-outputs_info = task.get_output_files_info()
-full_download_size = outputs_info.size # Size of the full output archive.
-files = outputs_info.contents # List of individual files available.
+# Get status of the task.
+>>> task.get_status()
+<DynamicSchema: 'success'>
+
+# Get the computational resource type where the task is
+# running or was submitted to
+>>> task.get_machine_type()
+'c2-standard-4'
+
+>>> task.get_computation_time()
+'00:03:12
```

Comment:
missing '

Btw, if this a string maybe we want to rethink. A string is something quite unhelpful because I can only print it. Maybe I want to have a "ytime" object or I want have:

task.get_computation_time_in_secs()

which returns an float.